pump/config_test.go

```diff
@@ -116,21 +117,46 @@ func (s *testConfigSuite) TestConfigParsingFileFlags(c *C) {
 	validateConfig(c, cfg)
 }
 
-func mustSuccess(c *C, err error) {
+func (s *testConfigSuite) TestConfigParsingFileWithInvalidArgs(c *C) {
+	yc := struct {
+		ListenAddr             string `toml:"addr" json:"addr"`
+		AdvertiseAddr          string `toml:"advertise-addr" json:"advertise-addr"`
+		EtcdURLs               string `toml:"pd-urls" json:"pd-urls"`
+		BinlogDir              string `toml:"data-dir" json:"data-dir"`
+		HeartbeatInterval      uint   `toml:"heartbeat-interval" json:"heartbeat-interval"`
+		UnrecognizedOptionTest bool   `toml:"unrecognized-option-test" json:"unrecognized-option-test"`
+	}{
+		"192.168.199.100:8260",
+		"192.168.199.100:8260",
+		"http://192.168.199.110:2379,http://hostname:2379",
+		"/tmp/pump",
+		1500,
+		true,
+	}
+
+	var buf bytes.Buffer
+	e := toml.NewEncoder(&buf)
+	err := e.Encode(yc)
 	c.Assert(err, IsNil)
-}
 
-func mustCreateCfgFile(c *C, b []byte, prefix string) *os.File {
-	tmpfile, err := ioutil.TempFile("", prefix)
-	mustSuccess(c, err)
+	configFilename := path.Join(c.MkDir(), "pump_config_invalid.toml")
+	err = ioutil.WriteFile(configFilename, buf.Bytes(), 0644)
+	c.Assert(err, IsNil)
 
-	_, err = tmpfile.Write(b)
-	mustSuccess(c, err)
+	args := []string{
+		"--config",
+		configFilename,
+		"-L", "debug",
+	}
 
-	err = tmpfile.Close()
-	mustSuccess(c, err)
+	os.Clearenv()
```

Comment:
Why do we need to clear environment variables before parsing again?

Reply:
It will be deleted in next commit.

Reply:
In `pump/config_test.go:75` test TestConfigParsingEnvFlags set environment variables. Now I add a defer clean job in this function. The other functions' os.Clearenv() have already been expurgated.